src/commands/deployImage/imageSource/containerRegistry/acr/AcrListStep.ts

```diff
@@ -69,3 +112,22 @@ export class AcrListStep extends AzureWizardPromptStep<IContainerRegistryImageCo
     }
 }
 
+async function tryConfigureResourceGroupForRegistry(
+    context: IContainerRegistryImageContext,
+    promptSteps: AzureWizardPromptStep<IContainerRegistryImageContext>[],
+): Promise<void> {
+    // No need to pollute the base context with all the potential pre-create typings as they are not otherwise used
+    const resourceCreationContext = context as Partial<ICreateContainerAppContext> & Partial<IManagedEnvironmentContext> & CreateAcrContext;
+    if (resourceCreationContext.resourceGroup || resourceCreationContext.newResourceGroupName) {
+        return;
+    }
+
+    // Try to leverage an existing resource group (managed environment name is a fallback here because our create flow has the resource group name mirror the environment name)
+    const resourceGroupName: string | undefined = resourceCreationContext.containerApp?.resourceGroup || resourceCreationContext.managedEnvironment?.name;
```

Comment:
I've been going back and forth on whether using the managed environment name as a search fallback makes sense or not, let me know if you have thoughts